cms/djangoapps/contentstore/signals.py

```diff
@@ -0,0 +1,19 @@
+""" receiver of course_published events in order to trigger indexing task """
+from datetime import datetime
+from pytz import UTC
+
+from django.dispatch import receiver
+
+from xmodule.modulestore.django import SignalHandler
+from contentstore.courseware_index import indexing_is_enabled
+
+
+@receiver(SignalHandler.course_published)
+def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receives signal and kicks off celery task to update search index
+    """
+    # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
+    from .tasks import update_search_index
+    if indexing_is_enabled():
+        update_search_index.delay(unicode(course_key), datetime.now(UTC))
```

Comment:
@martynjames actually, there might be another problem. I've just uploaded my #7641 to [sandbox](http://content-libraries.sandbox.opencraft.com:18010/) and it fails here:

    Sent course_published signal to <function listen_for_course_publish at 0x4951c08> with kwargs {'course_key': CourseLocator(u'Arbi', u'101', u'2015', None, None)}. Response was: datetime.datetime(2015, 4, 9, 16, 13, 55, 457562, tzinfo=<UTC>) is not JSON serializable

In two words, `datetime` objects are not json-serializable as well (not sure if it applies to all python versions though). So it might need to be converted to unicode and parsed back.